Makefile

```
@@ -57,24 +57,43 @@ build-all:
 aro: check-release generate
 	go build -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro
 
-runlocal-rp: ci-rp
-	@echo "Starting Podman service if not running..."
-	@podman machine init || true
-	@podman machine start || true
-	@echo "Podman service status:"
-	@podman machine list
-	@podman system connection list
-	@echo "Using ARO_IMAGE: ${ARO_IMAGE}"
-	@if [ -z "${ARO_IMAGE}" ]; then echo "Error: ARO_IMAGE is not set"; exit 1; fi
-	@podman run --rm -d -p 8443:8443 --name aro-rp ${ARO_IMAGE} rp
-	@echo "Podman logs for aro-rp container:"
-	@podman logs aro-rp || echo "Container aro-rp is not running"
+build-local-rp-image:
+	@echo "Increasing Podman machine resources..."
+	@podman machine stop
+	@podman machine set --cpus 4 --memory 8192
+	@podman machine start
+	@echo "Building local RP image..."
+	podman build . -f Dockerfile.ci-rp --ulimit=nofile=4096:4096 --build-arg REGISTRY=$(REGISTRY) --build-arg VERSION=$(VERSION) --no-cache=$(NO_CACHE) -t $(ARO_IMAGE) || true
+
+runlocal-rp:
+	set -a; source .env; set +a; \
+		$(MAKE) build-local-rp-image; \
+		echo "Starting Podman service if not running..."; \
+		podman machine init || true; \
+		podman machine start || true; \
+		echo "Podman service status:"; \
+		podman machine list; \
+		podman system connection list; \
+		echo "Using ARO_IMAGE: $(ARO_IMAGE)"; \
+		if [ -z "$(ARO_IMAGE)" ]; then echo "Error: ARO_IMAGE is not set"; exit 1; fi; \
+		podman run --rm -d -p 8443:8443 --name aro-rp -e DOMAIN_NAME=$(DOMAIN_NAME) -e AZURE_FP_CLIENT_ID=$(AZURE_FP_CLIENT_ID) $(ARO_IMAGE) rp; \
+	
+ensure-image:
```

IMO the target name could be clearer if it will be using `aro` in its name
```suggestion
ensure-aro-image:
```